core/src/main/java/org/jdbi/v3/core/result/internal/ResultSetResultIterator.java

```diff
@@ -70,15 +91,15 @@ public boolean hasNext() {
 
     @Override
     public T next() {
-        if (closed) {
-            throw new IllegalStateException("iterator is closed");
-        }
-
         if (!hasNext()) {
             close();
             throw new NoSuchElementException("No element to advance to");
         }
 
+        if (closed) {
```

Comment:
I think by moving this clause, it is made redundant. If the iterator is closed, `hasNext()` is false, so we already tried to `close(); throw ...;` above.
So if you want to change this, maybe just delete the clause instead?

Reply:
yes. Done.